services/stats/stats.go

```diff
@@ -42,14 +38,8 @@ type Stats interface {
 	NewTaggedStat(name, statType string, tags Tags) Measurement
 
 	// NewSampledTaggedStat creates a tagged measurement with the configured sampling rate (statsSamplingRate)
+	// Deprecated: use NewTaggedStat instead
 	NewSampledTaggedStat(name, statType string, tags Tags) Measurement
-
-	// Start starts the stats service. It tries to connect to statsd server once and returns immediately.
-	// If it fails to connect, it will retry in the background indefinitely, however any stats published in the meantime will be lost.
-	Start(ctx context.Context)
-
-	// Stop stops the service and collection of periodic stats.
-	Stop()
```

Comment:
It might be safer to follow a phased strategy like below:

1. Keep the old `statsd` implementation and keep using it as the default `stats` implementation.
2. Use a configuration option for being able to switch to the `otel` implementation so that we can test and verify it in a production environment.
3. As we become more confident we can make the `otel` implementation the default one and deprecate `statsd`.
4. Finally, remove support for statsd 
   - delete implementation
   - replace docker container with otel in docker compose file(s)
   - remove statsd environment variables etc.